src/helpers/folders.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Zextras <https://www.zextras.com>
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import { Account, Folder } from '@zextras/carbonio-shell-ui';
+import { find } from 'lodash';
+import { MailMessage } from '../types';
+
+/*
+ * Describe the folder id syntax
+ *
+ * [<zid>:]<folderId>
+ *
+ * e.g. a79fa996-e90e-4f04-97c4-c84209bb8277:2
+ */
+const FOLDERID_REGEX = /^([^:]+(?=:))*:?(\d+)$/;
+
+type FolderIdType = { zid: string | undefined; id: string | undefined };
+
+/**
+ * Parse the given folder id and returns on object with the composing parts of the folder id
+ * @param folderId
+ */
+const getFolderIdParts = (folderId: string): FolderIdType => {
+	const result: FolderIdType = { zid: undefined, id: undefined };
+
+	if (!folderId || !folderId.match(FOLDERID_REGEX)) {
+		return result;
+	}
+
+	const parts = FOLDERID_REGEX.exec(folderId);
+	if (!parts) {
+		return result;
+	}
+
+	[, result.zid, result.id] = parts;
+	return result;
+};
+
+/**
+ * Get the account name of the owner of the given folder, if the owner is an
+ * "other" account, different from the primary account of the current user.
+ * If the owner is the primary account then <code>undefined</code> is returned
+ * @param folderId
+ * @param folderRoots
+ */
+const getFolderOtherOwnerAccountName = (
+	folderId: string,
+	folderRoots: Record<string, Folder & { owner: string }>
+): string | undefined => {
+	if (!folderId) {
+		return undefined;
```

Comment:
I generally would return null rather than undefined. undefined is used by javascript when a value has not been assigned, null indicates that something definitely has no value

Reply:
You're right. _Null_ is semantically clearer. I'll change it